### PR TITLE
Mech punches produce attack logs.

### PR DESCRIPTION
**Pull request by mikomyazaki**

### CODE REVIEW

File: code/modules/mechs/mech_interaction.dm [hunk 1]
```diff
@@ -171,6 +171,9 @@
 		return attack_self(user)
 	else if(adj)
 		setClickCooldown(arms ? arms.action_delay : 15)
+		if(ismob(A))
+			var/mob/target = A
```
Comment (afterthought2) [inline code/modules/mechs/mech_interaction.dm]: Don't think you need this cast.

Comment (MistakeNot4892) [reply code/modules/mechs/mech_interaction.dm]: Should be handled by the arg types in the proc call.

**Review by MistakeNot4892 — DISMISSED** [review]

File: code/modules/mechs/mech_interaction.dm [hunk 2]
```diff
@@ -171,6 +171,7 @@
 		return attack_self(user)
 	else if(adj)
 		setClickCooldown(arms ? arms.action_delay : 15)
+		admin_attack_log(user, target, "Attacked using \a [arms]", "Was attacked with \a [arms]", "used \a [arms] to attack")
```
Comment (MistakeNot4892) [inline code/modules/mechs/mech_interaction.dm]: target => A

Comment (mikomyazaki) [reply code/modules/mechs/mech_interaction.dm]: Oops.

**Review by MistakeNot4892 — APPROVED** [review]

**Pull request merged**